src/os/bluestore/BlueStore.cc

```diff
@@ -11776,11 +11841,131 @@ int BlueStore::read(
     if (offset == length && offset == 0)
       length = o->onode.size;
 
-    r = _do_read(c, o, offset, length, bl, op_flags);
+    r = _do_read(c, o, offset, length, bl, op_flags, 0,
+          might_need_reformatting ? &span_stat : nullptr);
```

Comment:
If we can potentially defrag on any read, it could be interesting once the extents are read off disk to make a decision about whether or not an object is fragmented enough to warrant defragmentation on-the-fly.

Reply:
@markhpc I am excited to see it. Opens interesting paths.